### PR TITLE
H-308: Enforce generation of system type TS types as part of CI

### PR DESCRIPTION
**Pull request by benwerner01**
## 🔗 Related links

- H-308

## 🚫 Blocked by

- [ ] ...

## 🔍 What does this change?

- ...

### 🚢 Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing
- [x] modifies an **npm**-publishable library and **I have added a changeset file(s)**
- [x] modifies a **Cargo**-publishable library and **I have amended the version**
- [x] modifies a **Cargo**-publishable library, but **it is not yet ready to publish**
- [x] modifies a **block** that will need publishing via GitHub action once merged
- [x] I am unsure / need advice

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] are internal and do not require a docs change
- [x] are in a state where docs changes are not _yet_ required but will be
  - this is tracked in: [Insert Link Here](link)
- [x] require changes to docs which **are made** as part of this PR
- [x] require changes to docs which are **not** made in this PR
  - _Provide more detail here_
- [x] I am unsure / need advice

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] do not affect the execution graph
- [x] affected the execution graph, and the `turbo.json`'s have been updated to reflect this
- [x] I am unsure / need advice

## 🛡 What tests cover this?

-

## ❓ How to test this?

1. Checkout the branch / view the deployment
1. Try X
1. Confirm that Y

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -370,6 +370,71 @@ jobs:
...
+        run: |
+          yarn workspace @apps/hash-frontend next build
+          yarn workspace @apps/hash-frontend start 2>&1 | tee var/logs/frontend.log & ## ampersand enables background mode
+          yarn wait-on --timeout 30000 http://0.0.0.0:3000
+
+      - name: Generate system types
+        env:
+          FRONTEND_URL: http://0.0.0.0:3000
+        run: |
+          yarn wait-on --timeout 30000 http://0.0.0.0:3000
```
Comment (TimDiekmann) [inline .github/workflows/ci.yml:423]: I don't know why the script does not work as expected but this line is the same as the last line in the previous step so it should do nothing.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -370,6 +370,71 @@ jobs:
...
+
+      - name: Build and launch frontend
+        run: |
+          yarn workspace @apps/hash-frontend next build
+          yarn workspace @apps/hash-frontend start 2>&1 | tee var/logs/frontend.log & ## ampersand enables background mode
+          yarn wait-on --timeout 30000 http://0.0.0.0:3000
+
+      - name: Generate system types
+        env:
+          FRONTEND_URL: http://0.0.0.0:3000
```
Comment (TimDiekmann) [inline .github/workflows/ci.yml:421]: Did you try localhost instead?
```suggestion
          FRONTEND_URL: http://127.0.0.1:3000
```

**Review by TimDiekmann — COMMENTED** [review]
To check if the generation needs to run you can check for
```
turbo run build --filter "@local/hash-isomorphic-utils...[HEAD^]" --dry-run=json | jq -r '.packages != []'
```
in the setup job.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -370,6 +370,71 @@ jobs:
...
+        run: |
+          yarn wait-on --timeout 30000 http://0.0.0.0:3000
+          yarn generate-system-types
+
+      - name: Check for changes in system-types directory
+        run: |
+          if ! git diff --exit-code apps/hash-api/src/graph/knowledge/system-types; then
```
Comment (TimDiekmann) [inline .github/workflows/ci.yml:428]:
```suggestion
          if ! git --no-pager diff --exit-code --color apps/hash-api/src/graph/knowledge/system-types; then
```

File: .github/workflows/ci.yml (hunk 2 above)
Comment (benwerner01) [reply .github/workflows/ci.yml]: Unfortunately `http://127.0.0.1:3000` doesn't seem to have worked either (see [this run](https://github.com/hashintel/hash/actions/runs/6237832507/job/16932377540?pr=3199))

**Pull request closed without merging**